bitbucket-slack-server-integration-plugin/pom.xml

```diff
@@ -5,7 +5,7 @@
     <groupId>com.atlassian.bitbucket.plugins</groupId>
     <artifactId>bitbucket-slack-server-integration-plugin</artifactId>
     <packaging>atlassian-plugin</packaging>
-    <version>3.0.5-SNAPSHOT</version>
+    <version>3.1.0-SNAPSHOT</version>
```

Comment:
Minor version upgrade is usually performed during the release process and not as a PR change. Additionally, minor version is changed when some new features are added. I don't see any new features introduced in this PR. Could you revert this change?

Reply:
If that's how you folks work, fine. But Manish is doing the right thing here. 

This pull request is introducing changes that are incompatible with versions of BbS that this plugin currently supports (no new features, but still incompatible changes). Semver says he should, at least, increment the minor version number.

In addition, he's making the version change in this pull request so the merge commit that makes the incompatible changes also contains a version bump indicating that they're incompatible changes... it's a bit like self-documenting code. Otherwise, there's a chance that someone thinks "the new version is incompatible with my version of Bitbucket - I'll branch from before the version was changed to pick up compatible changes" but they'll end up getting these changes anyway.

I understand that it's subtle, but it's definitely how/when this kind of version bump should be done.

Reply:
Reverted the minor version bump

Reply:
We don't have a strict process on this because SEMVER is effectively meaningless for plugins since they are not published in any repository but manually market compatible in the Atlassian Marketplace.

Having said that, I think it would make sense bumping the version here. During the release process, we would just leave the release version as is instead of overriding it. Or maybe I'm unaware about something specific about our GitHub setup. 🤷‍♂️